ocaml/typing/typecore.ml

```diff
@@ -3021,7 +3008,6 @@ let rec check_counter_example_pat
       end
   | Tpat_alias (p, _, _, _, _) -> check_rec ~info p expected_ty k
   | Tpat_constant cst ->
-      let cst = constant_or_raise !env loc (Untypeast.constant cst) in
```

Comment:
Why is it ok to delete this line?

Reply:
Although I'm relatively confident that the constant here is always going to be well-formed (since we typed it once already) and we will just be getting the same thing back again from `constant_or_raise`, I did add the logic back in after the suggested refactor of `Untypeast.constant` which makes it possible.